app/assets/javascripts/admin/models/flagged-post.js.es6

```diff
@@ -62,11 +62,74 @@ export default Post.extend({
   },
 
   deferFlags(deletePost) {
-    return ajax('/admin/flags/defer/' + this.id, { type: 'POST', cache: false, data: { delete_post: deletePost } }).catch(popupAjaxError);
+    const action = () => {
+      return ajax('/admin/flags/defer/' + this.id, {
+        type: 'POST', cache: false, data: { delete_post: deletePost }
+      });
+    };
+
+    if (deletePost && this._hasDeletableReplies()) {
+      return this._actOnFlagAndDeleteReplies(action);
+    } else {
+      return action().catch(popupAjaxError);
+    }
   },
 
   agreeFlags(actionOnPost) {
-    return ajax('/admin/flags/agree/' + this.id, { type: 'POST', cache: false, data: { action_on_post: actionOnPost } }).catch(popupAjaxError);
+    const action = () => {
+      return ajax('/admin/flags/agree/' + this.id, {
+        type: 'POST', cache: false, data: { action_on_post: actionOnPost }
+      });
+    };
+
+    if (actionOnPost === 'delete' && this._hasDeletableReplies()) {
+      return this._actOnFlagAndDeleteReplies(action);
+    } else {
+      return action().catch(popupAjaxError);
+    }
+  },
+
+  _hasDeletableReplies() {
+    return this.get('post_number') > 1 && this.get('reply_count') > 0;
+  },
+
+  _actOnFlagAndDeleteReplies(action) {
+    return new Ember.RSVP.Promise((resolve, reject) => {
+      return ajax(`/posts/${this.id}/reply-ids/all.json`).then(replies => {
```

Comment:
I think we should avoid this extra request and do it all on the server side. We can use`reply_count` to replace `replies.length`.

Reply:
`reply_count` is only the number of direct replies and `replies.length` is the number of all replies, so replacing won't work. I guess I could replace the second call by sending the reply ids to the server.